kernels/simple_matmul/main.c

```diff
@@ -44,103 +44,53 @@ uint32_t strideC = 0;
 
-void _mlir_ciface_simple_matmul_cpu(TwoDMemrefI8_t *a, TwoDMemrefI8_t *b,
-                                    TwoDMemrefI32_t *c) {
+void _mlir_ciface_snax_qgemm(TwoDMemrefI8_t *a, TwoDMemrefI8_t *b, int32_t zpa,
+                             int32_t zpb, TwoDMemrefI32_t *c) {
+
   int8_t *a_ptr = a->aligned_data;
   int8_t *b_ptr = b->aligned_data;
   int32_t *c_ptr = c->aligned_data;
-  batch_gemm_cpu(Batch, M_param, K_param, N_param, a_ptr, b_ptr, 0, 0, c_ptr,
-                 strideInnermostA, strideInnermostB, strideInnermostC, ldA, ldB,
-                 ldC, strideA, strideB, strideC);
+  printf("Executing snax_qgemm with a=%p, b=%p, c=%p \n", a_ptr, b_ptr, c_ptr);
+
+  uint32_t size_setting = gen_size_config(Batch, M_param, K_param, N_param);
+
+  set_batch_gemm(size_setting, a_ptr, b_ptr, 0, c_ptr, strideInnermostA,
+                 strideInnermostB, strideInnermostC, ldA, ldB, ldC, strideA,
+                 strideB, strideC);
+
+  start_batch_gemm();
+
+  wait_batch_gemm();
+
+  printf("Finished executing snax_qgemm\n");
 }
 
 int main() {
-  // Allocate space in TCDM
-  // We put the data in different banks, but we don't interleave the data for
-  // now.
-  //
-  //  | A | x | x | x |  --> A in banks 0 - 7  --> (8/32 banks used)*
-  //                                               (int8 --> 8 elements/bank)
-  //                                               1 row --> 64 elements
-  //  | x | B | x | x |  --> B in banks 7 - 15 --> (8/32 banks used)*
-  //                                               (8 elements/bank)*32 banks
-  //                                               1 row --> 64 elements
-  //  | C | C | C | C |  --> C in banks 0 - 31 --> (32/32 banks used)*
-  //                                               (2 elements/bank)* 32 bank
-  //                                               1 row --> 64 elements
-  //  | x | x | x | x |
-  //
-  //  32 banks -->  1 row = 32 banks * 8 bytes --> 256 adresses further
-
-  static int8_t *allocated_a;
-  static int8_t *allocated_b;
-  static int32_t *allocated_c;
-
-  // Transfer data from L3 to L1
-  // Using DMA only
-  if (snrt_is_dm_core()) {
-    // calculation in bytes directly
-    allocated_a = (int8_t *)snrt_l1alloc(256 * M_size * K_size / 64);
-    allocated_b = (int8_t *)snrt_l1alloc(256 * K_size * N_size / 64);
-    allocated_c = (int32_t *)snrt_l1alloc(256 * M_size * N_size / 64);
-  }
-  snrt_cluster_hw_barrier();
 
-  // Create memref descriptors for data stored in L1
+  // Create memref objects for data stored in L3
   TwoDMemrefI8_t memrefA;
-  memrefA.data = allocated_a;
+  memrefA.data = &A;
   memrefA.aligned_data = memrefA.data;
-  memrefA.shape[0] = M_size;
-  memrefA.shape[1] = K_size;
-  // The following values of this memref are ignored right now.
-  // A 2D memref is not enough to express a tiled-block layout (=4D),
-  // necessary by the accelerator,
-  // Instead we use the variables strideInnermostA, ldA and strideA
```

Comment:
These values still seem to be ignored? Can we leave this comment here in some form maybe?

Reply:
ok i'll check i maybe removed too much